lib/host.py

```diff
@@ -85,6 +85,32 @@ def stringify(key, value):
             return False
         return result
 
+    def create_file(self, filename, text):
+        with tempfile.NamedTemporaryFile('w') as file:
+            file.write(text)
+            file.flush()
+            self.scp(file.name, filename)
+
+    def add_xcpng_repo(self, name):
+        content = """\
+[xcp-ng-{}]
+name=XCP-ng {} Repository
+baseurl=http://mirrors.xcp-ng.org/{}/{}/{}/x86_64/ http://updates.xcp-ng.org/{}/{}/{}/x86_64/
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-xcpng
+"""
+
+        major = self.xcp_version.major
+        self.create_file(
+            '/etc/yum.repos.d/xcp-ng-{}.repo'.format(name),
+            content.format(name, name, major, self.xcp_version, name, major, self.xcp_version, name)
+        )
```

Comment:
Not a blocking comment, but I would find it much more readable with a f-string.

Reply:
Right. I forget it's not python2. :)